cmd/add-user-directories.go

```diff
@@ -0,0 +1,137 @@
+// Package cmd implements access-cli commands
+package cmd
+
+/*
+Copyright © 2020 Barracuda Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"github.com/spf13/cobra"
+
+	api "github.com/barracuda-cloudgen-access/access-cli/client/user_directories"
+	"github.com/barracuda-cloudgen-access/access-cli/models"
+)
+
+// userDirectoriesAddCmd represents the add command
+var userDirectoriesAddCmd = &cobra.Command{
+	Use:     "add",
+	Aliases: []string{"create", "new"},
+	Short:   "Add user directories",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		err := preRunCheckAuth(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = preRunFlagChecks(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		tw := userDirectoryBuildTableWriter()
+		createdList := []*models.UserDirectory{}
+		total := 0
+
+		err := forAllInput(cmd, args, true,
+			func(values *inputEntry) (interface{}, error) { // do func
+				total++ // this is the total of successful+failures, must increment before failure
+				userdirectory := &api.CreateUserDirectoryParamsBodyUserDirectory{}
+
+				err := placeInputValues(cmd, values, userdirectory,
+					func(s string) { userdirectory.ShortCode = s },
+					func(s string) { userdirectory.Name = s },
+					func(s string) { userdirectory.DirectoryType = s },
+					func(s string) { userdirectory.Notes = s })
+				if err != nil {
+					return nil, err
+				}
+				body := api.CreateUserDirectoryBody{UserDirectory: userdirectory}
+				params := api.NewCreateUserDirectoryParams()
+
+				setTenant(cmd, params)
+				params.SetUserDirectory(body)
+
+				resp, err := global.Client.UserDirectories.CreateUserDirectory(params, global.AuthWriter)
+				if err != nil {
+					return nil, err
+				}
+				return resp.Payload.UserDirectory, nil
+			}, func(data interface{}) { // printSuccess func
+				userdirectory := data.(models.UserDirectory)
```

Comment:
this response has the `enrollment_url` on it that is only returned on creation, I think you should print that separately from the table with a message or something saying "you put this on the connector", whatever we are doing in the proxy add flow (or if we are not doing it there we should also fix that)